vault/identity_store_oidc_provider.go

```diff
@@ -161,3 +216,129 @@ func (i *IdentityStore) pathOIDCAssignmentExistenceCheck(ctx context.Context, re
 
 	return entry != nil, nil
 }
+
+// pathOIDCCreateUpdateScope is used to create a new scope or update an existing one
+func (i *IdentityStore) pathOIDCCreateUpdateScope(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+
+	var scope scope
+	if req.Operation == logical.UpdateOperation {
+		entry, err := req.Storage.Get(ctx, scopePath+name)
+		if err != nil {
+			return nil, err
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(&scope); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if descriptionRaw, ok := d.GetOk("description"); ok {
+		scope.Description = descriptionRaw.(string)
+	} else if req.Operation == logical.CreateOperation {
+		scope.Description = d.GetDefaultOrZero("description").(string)
+	}
+
+	if templateRaw, ok := d.GetOk("template"); ok {
+		scope.Template = templateRaw.(string)
+	} else if req.Operation == logical.CreateOperation {
+		scope.Template = d.GetDefaultOrZero("template").(string)
+	}
+
+	// Attempt to decode as base64 and use that if it works
+	if decoded, err := base64.StdEncoding.DecodeString(scope.Template); err == nil {
+		scope.Template = string(decoded)
+	}
+
+	// Validate that template can be parsed and results in valid JSON
+	if scope.Template != "" {
+		_, populatedTemplate, err := identitytpl.PopulateString(identitytpl.PopulateStringInput{
```

Comment:
I see that this is exactly similar to the template check for identity token roles. I'm wondering if using [ValidityCheckOnly](https://github.com/hashicorp/vault/blob/main/sdk/helper/identitytpl/templating.go#L29) makes sense here. Otherwise, [templating](https://github.com/hashicorp/vault/blob/main/sdk/helper/identitytpl/templating.go#L151) is performed. I'll look at this a bit closer.

I'm also curious about the `// namespace?` comment. Looking into that as well.